src/engine/ExportQueryExecutionTrees.h

```diff
@@ -178,16 +178,20 @@ class ExportQueryExecutionTrees {
       const parsedQuery::SelectClause& selectClause,
       LimitOffsetClause limitAndOffset, CancellationHandle cancellationHandle);
 
-  struct IndexWithTable {
-    size_t index_;
+  // Helper type that contains an `IdTable` and a view with related indices to
+  // access the `IdTable` with.
+  struct TableWithRange {
     const IdTable& idTable_;
+    std::ranges::iota_view<uint64_t, uint64_t> view_;
```

Comment:
Do you need the original absolute indices into the IdTable?
Otherwise you could return a single
`std::ranges::subrange<const IdTable&>` 
and all your loops become singular again, and all the sonarcloud stuff goes away?

Reply:
I need the values as well as a reference to the original `IdTable` to fill in the `ConstructQueryExportContext`

Reply:
It would be possible however, to refactor `ConstructQueryExportContext` to get some row reference instead, but this might be better suited for a follow-up PR.

Reply:
Thinking about it it doesn't solve the original issue. yielding subranges still requires 2 loops